dist/bin/dotc

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+
+if [ -z "$PROG_HOME" ] ; then
```

Comment:
There's a lot of copy paste and re-implementation for the current `bin/dotc` script.
I propose we merge the two, otherwise it will be hard to maintain and we'll be using different scripts from the ones our users do.